docs/source/usage.rst

```diff
@@ -458,6 +458,34 @@ This can be useful for instance to parse environment variables:
     type: int, value: 3308
 
 
+Extracting lists of keys / values from a dictionary
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some config options that are stored as a ``DictConfig`` may sometimes be easier to manipulate as lists,
+when we care only about the keys or the associated values.
+
+The resolvers ``oc.dict.keys`` and ``oc.dict.values`` simplify such operations by extracting respectively
+the list of keys and values from ``dict``-like objects like ``DictConfig``:
+
+.. doctest::
+
+    >>> cfg = OmegaConf.create(
+    ...     {
+    ...         "machines": {
+    ...             "node007": "10.0.0.7",
+    ...             "node012": "10.0.0.3",
+    ...             "node075": "10.0.1.8",
+    ...         },
+    ...         "nodes": "${oc.dict.keys:${machines}}",
+    ...         "ips": "${oc.dict.values:${machines}}",
```

Comment:
What do you think about using select syntax instead?

```suggestion
    ...         "nodes": "${oc.dict.keys:machines}",
    ...         "ips": "${oc.dict.values:machines}",
```
Seems significantly cleaner.

Reply:
Can do, only thing is it makes it a bit less powerful (actually restricting it to existing nodes of the config, while the current implementation could be used on any dict-like object, e.g. coming out of another custom resolver).

Reply:
> Something else: I just remembered that I have a WIP change that reverts the automatic conversion of list/dict resolver outputs, relying on explicitly using `oc.create` instead (relevant discussion: [#606 (comment)](https://github.com/omry/omegaconf/pull/606#issuecomment-800589117))
> 
> Do you prefer that `oc.dict.keys` output plain dicts / lists (enforcing the usage of `oc.create` to convert them to config containers), or that they output directly DictConfig / ListConfig (the latter is a bit less flexible but probably better for most use cases, so I'm leaning towards it)

The motivating use case for this is to generate Lists config nodes from dict config nodes, so let's convert to ListConfig automatically.
if someone wants something else they can register their own resolver for now.

Reply:
> Can do, only thing is it makes it a bit less powerful (actually restricting it to existing nodes of the config, while the current implementation could be used on any dict-like object, e.g. coming out of another custom resolver).
> 
> Unless we support both?

I was thinking both.
See my comment about supported input types.

Reply:
Done in 872eb2d